### PR TITLE
Remove cost_price and other dead code

### LINKED ISSUE #6898: Removal other code bits related to "cost price" field
### What we should change and why (this is tech debt)

This issue refers to the removal of code bits related to the "cost price" field. This includes the var `gross_profit` and our current `db/schema.rb`, namely Spree::Variant:

 

### Context

This is a followup on PR #6822. See testing notes and discussion.

### Impact and timeline

It aims to simplify future PRs - for example, we probably should not need to consider this field when making changes on the DB:

### PR DESCRIPTION
**Pull request by Matt-Yorkley**
#### What? Why?

Closes #6898

Removes cost_price fields from variants and line_items, and deletes related code.

Also removes several other bits of dead code found at the same time. :fire:

#### What should we test?

Green build.

#### Release notes

Removed variant cost_price and related code

Changelog Category: Technical changes

### CODE REVIEW

**Comment by Matt-Yorkley** [comment]
I'm starting to feel like the kid in Sixth Sense...

I-see-dead-code

**Review by sauloperez — APPROVED** [review]
:fire: :fire:

**Comment by Matt-Yorkley** [comment]
Rebased :+1:

File: app/models/spree/variant.rb [hunk 1]
```diff
@@ -168,63 +167,17 @@ def fees_by_type_for(distributor, order_cycle)
...
     # use deleted? rather than checking the attribute directly. this
     # allows extensions to override deleted? if they want to provide
     # their own definition.
     def deleted?
       deleted_at
     end
 
-    def set_option_value(opt_name, opt_value)
-      # no option values on master
-      return if is_master
```
Comment (andrewpbrett) [inline app/models/spree/variant.rb:189]: I could have sworn this was the method that was getting called when the various [nil variant migrations](https://github.com/openfoodfoundation/openfoodnetwork/pull/6841) were failing, but I took another look at them and I think I'm misremembering. We're sure it's dead code? I don't see that string `set_option_value` anywhere but here and the spec, so it definitely seems dead. 

https://gifimage.net/wp-content/uploads/2017/11/im-not-dead-yet-gif-1.gif

**Review by andrewpbrett — APPROVED** [review]

File: app/models/spree/variant.rb (hunk 1 above)
Comment (Matt-Yorkley) [reply app/models/spree/variant.rb:189]: It looks like it was used in an old Spree method in the Spree::Variant class, which we removed.

**Pull request merged**